core/src/main/scala/com/typesafe/tools/mima/core/TastyUnpickler.scala

```diff
@@ -101,6 +102,33 @@ object TastyUnpickler {
     }
   }.traverse(tree)
 
+  def copyClassPrivate(tree: Tree, pkgInfo: PackageInfo): Unit = new ClassTraverser(pkgInfo) {
+    override def forEachClass(clsDef: ClsDef, cls: ClassInfo): Unit = {
+    }
+
+    override def traverseTemplate(tmpl: Template): Unit = {
+      super.traverseTemplate(tmpl)
+      doMethods(tmpl)
+    }
+
+    def doMethods(tmpl: Template) = {
+      val clazz = currentClass
+      tmpl.meths.iterator
+        .toSeq.groupBy(_.name).foreach { case (name, pickleMethods) =>
+          doMethodOverloads(clazz, name, pickleMethods)
+        }
+    }
+
+    def doMethodOverloads(clazz: ClassInfo, name: Name, pickleMethods: Seq[DefDef]) = {
+      val bytecodeMethods = clazz.methods.get(name.source).filter(!_.isBridge).toList
+      if (pickleMethods.size == bytecodeMethods.size && pickleMethods.exists(_.classPrivate)) {
+        bytecodeMethods.zip(pickleMethods).foreach { case (bytecodeMeth, pickleMeth) =>
+          bytecodeMeth.classPrivate = pickleMeth.classPrivate
+        }
+      }
+    }
+  }.traverse(tree)
```

Comment:
Could you please explain why it is not possible to do the job in `copyPrivateWithin`? I am a bit lost.